protocol_test.go

```diff
@@ -68,19 +68,23 @@ func BenchmarkCanonicalizeContentType(b *testing.B) {
 func TestProcedureFromURL(t *testing.T) {
 	t.Parallel()
 	tests := []struct {
-		name string
-		url  string
-		want string
+		name  string
+		url   string
+		want  string
+		valid bool
```

Comment:
Need an assertion for this (which would have caught what I assume is a bug pointed out by other comment)

Reply:
Sorry, not sure how this slipped up. I think I was adding this testcase and missed the commit. Have fixed.